test/state/state.cpp

```diff
@@ -162,7 +164,16 @@ std::variant<TransactionReceipt, std::error_code> transition(
     if (rev >= EVMC_SHANGHAI)
         host.access_account(block.coinbase);
 
-    const auto result = host.call(build_message(tx, execution_gas_limit));
+    auto is_valid_tx = true;
+    auto result = evmc::Result{};
+
+    if (rev >= EVMC_CANCUN && !tx.to.has_value() && is_eof_code(tx.data))
+        is_valid_tx = (validate_eof(rev, tx.data) == evmone::EOFValidationError::success);
```

Comment:
I guess this has to change with recent changes? (Bad initcode is valid tx.)

Reply:
Exactly. We have to decide which version we want to have implemented here.